python/ray/dag/compiled_dag_node.py

```diff
@@ -381,18 +409,22 @@ def _get_or_compile(
             visited.add(cur_idx)
 
             task = self.idx_to_task[cur_idx]
-            # Create an output buffer on the actor.
+            # Create an output buffer for the actor method.
             assert task.output_channel is None
             if isinstance(task.dag_node, ClassMethodNode):
                 fn = task.dag_node._get_remote_method("__ray_call__")
+                method_name = task.dag_node.get_method_name()
+                logger.info("Creating output channel for method %s", method_name)
                 task.output_channel = ray.get(
                     fn.remote(
                         do_allocate_channel,
                         buffer_size_bytes=self._buffer_size_bytes,
                         num_readers=task.num_readers,
                     )
                 )
-                self.actor_refs.add(task.dag_node._get_actor_handle())
+                actor_handle = task.dag_node._get_actor_handle()
+                self.actor_to_ref[actor_handle._actor_id] = actor_handle
+                self.actor_to_tasks[actor_handle._actor_id].append(task)
```

Comment:
Does this follow the ordering of the task submission? I.e.

```python
with InputNode as inp:
  dag = a.foo.bind(inp)
  dag = a.bar.bind(dag)
```

Are we guaranteed that a.bar will execute after a.foo? This isn't a great example since there's actually a data dependency between the tasks, but the idea here is to enforce a control dependency on tasks bound to the same actor based on the order of creation in the DAG.

Reply:
Fixed and added a unit test.